### PR TITLE
Run Quinn tests in CI

### LINKED ISSUE #6717: Improve testing of timers
A regression in the timer code was discovered in #6714. It was not caught by our test-suite, so we should improve it.

Suggested solution: Add quinn to CI, as quinn's tests caught the bug.

### PR DESCRIPTION
**Pull request by djc**
## Motivation

Improve testing for timers (which caused a regression, #6714) by testing Quinn in CI.

Fixes #6717.

### CODE REVIEW

**Comment by djc** [comment]
Not sure how/why warnings are being denied, probably makes sense to allow them for these external crates?

**Comment by Darksonn** [comment]
You probably need to override RUST_FLAGS.

https://github.com/tokio-rs/tokio/blob/f8fe0ffb23c2279708cb5dada7c88defdae60845/.github/workflows/ci.yml#L14

**Comment by Darksonn** [comment]
Any chance you could try pushing a commit that undoes the revert to verify that this would have caught the regression?

**Comment by djc** [comment]
I used

```
djc-2021 quinn-ci tokio $ git cherry-pick 8480a180e6ffdd0ec0ec213a9bebcda2445fc541
```

and that resulted in [failure](https://github.com/tokio-rs/tokio/actions/runs/10091499773/job/27903242161?pr=6719).

So I think this is working as intended, and I'll remove the cherry-picked commit again.

File: .github/workflows/ci.yml
Comment (Darksonn) [inline .github/workflows/ci.yml:1]: Are you able to undo the formatting changes in this file?

**Review by Darksonn — APPROVED** [review]
Awesome, thank you. Verified that this would have caught the regression.

File: .github/workflows/ci.yml
Comment (djc) [reply .github/workflows/ci.yml:1]: Dropped that commit.

**Comment by Darksonn** [comment]
Thanks!

**Comment by djc** [comment]
Thank you for all your work on Tokio!

**Pull request merged**